java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java

```diff
@@ -114,10 +114,14 @@ public static String trimLongSuffix(String longString) {
   }
 
   public static boolean hasValue(Tree tree, String expectedValue) {
-    if (!tree.is(Tree.Kind.STRING_LITERAL)) {
+    if (!tree.is(Kind.STRING_LITERAL)) {
       return false;
     }
-    String mechanismName = trimQuotes(((LiteralTree) tree).value());
-    return expectedValue.equals(mechanismName);
+    String actualValue = trimQuotes(((LiteralTree) tree).value());
+    return expectedValue.equals(actualValue);
+  }
+
+  public static boolean hasValue(Tree tree, boolean expectedValue) {
+    return tree.is(Kind.BOOLEAN_LITERAL) && expectedValue == (Boolean.parseBoolean(((LiteralTree) tree).value()));
```

Comment:
Why not comparing strings rather than parsing the literal value?

Reply:
Because I am passing a boolean parameter.